ml-agents/mlagents/trainers/trainer/trainer_factory.py

```diff
@@ -100,8 +104,6 @@ def _initialize_trainer(
         :return:
         """
         trainer_artifact_path = os.path.join(output_path, brain_name)
-        if init_path is not None:
-            trainer_settings.init_path = os.path.join(init_path, brain_name)
```

Comment:
Moved this logic to learn.py

Reply:
Have you tried using init_path and see if it does what we expect out of it?
Reading a bit of the code in `torch_model_saver.py` it looks like setting init_path in the trainer settings initializes the policy from a checkpoint. Is that correct?

Reply:
Correct. Previously `trainer_settings.init_path="result_dir/run_id/brain_name` and we added the checkpoint name at the end in `torch_model_saver.py`
This PR sets the full path `trainer_settings.init_path="result_dir/run_id/brain_name/checkpoint_name.pt` in [learn.py:102](https://github.com/Unity-Technologies/ml-agents/blob/b0c2361031af87551dc7ecf5595a43020651e09e/ml-agents/mlagents/trainers/learn.py#L102)